mlos_bench/mlos_bench/dict_templater.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Simple class to help with nested dictionary $var templating.
+"""
+
+from copy import deepcopy
+from string import Template
+from typing import Any, Dict, Optional
+
+import os
+
+
+class DictTemplater:    # pylint: disable=too-few-public-methods
+    """
+    Simple class to help with nested dictionary $var templating.
+    """
+
+    def __init__(self, source_dict: Dict[str, Any]):
+        """
+        Initialize the templater.
+
+        Parameters
+        ----------
+        source_dict : Dict[str, Any]
+            The template dict to use for source variables.
+        """
+        # A copy of the initial data structure we were given with templates intact.
+        self._template_dict = deepcopy(source_dict)
+        # The source/target dictionary to expand.
+        self._dict: Dict[str, Any] = {}
+
+    def expand_vars(self, *,
```

Comment:
we have only two parameters of different types, so maybe there is no need to enforce explicit kwargs
```suggestion
    def expand_vars(self,
```
I'm ambivalent on this one, though

Reply:
Leaving that one as is for my own readability for now.